### DISCUSSION TITLE
Sleep-On-LAN (SOL)

### DISCUSSION THREAD

**Discussion by bkrisz14** [AlexGustafsson/homebridge-wol#216] [Questions]
### General idea

Is it possible to write a command for sending the PC to sleep mode instead of shut down?

### Use case

I'm using sleep mode more often than shutting down the computer.

**Comment by AlexGustafsson** [comment]
Yes it is possible, however, there is no "Sleep On LAN" the same way that there's Wake On LAN, so this package cannot support all use cases automatically. However, you can specify any shutdown command to enable this to work. You will have to install some custom software on the target PC and simply make your shutdown command use whatever means that program exposes to turn the target off.

Some examples:

- https://github.com/SR-G/sleep-on-lan
- https://github.com/AlexGustafsson/homebridge-wol/wiki/Running-on-Synology-Docker-and-turn-Windows-10-on-off-using-SSH-and-Samba
- https://www.wikihow.com/Remotely-Shutdown-a-Computer

**Comment by alissaman** [comment]
I got sleep to work using a combination of Open SSH on windows to call psshutdown.exe which was copied to where SSH starts in as I needed sleep rather than shutdown. 

https://learn.microsoft.com/en-us/sysinternals/downloads/psshutdown

The command I used after configuring RSA keys for auth is:

ssh user@192.168.1.10 psshutdown64.exe -d -t 1

-d = sleep
-t 1 = 1 second wait before sleep
